764 feature html export

BobTheSuperman opened the pull request.

BobTheSuperman committed:
Added ExportService for html file export, and dependencies for it to work. Also added files for working with this service.

BobTheSuperman committed:
Added ability to Export students classbook and student groups in html format

AccorDDD commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+{
+    class ClassbookExportHTML : BaseFileExportHTML
+    {
+        public ClassbookExportHTML()
+        {
+
+        }
```
Why is this constructor here?

BobTheSuperman replied:
Removed it, ty.

ArsenMaistryshyn commented on CharlieBackend.Business/Helpers/HtmlGenerator.cs:
```diff
@@ -0,0 +1,72 @@
+            table.Append(tableBody);
+            table.Append("</table>");
+
+
```
please remove space

BobTheSuperman replied:
Done!

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/ExportServiceHtml.cs:
```diff
@@ -0,0 +1,93 @@
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using System.Threading.Tasks;
+
```
please remove space

BobTheSuperman replied:
Done!

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/ExportServiceHtml.cs:
```diff
@@ -0,0 +1,93 @@
+            else
+            {
+                return Result<FileDto>.GetError(ErrorCode.NotFound,
+                        "Group not found");
```
please extract to response messages

BobTheSuperman replied:
Done

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+            }
+        }
+
+        private void TryToFillMarks(StudentsClassbookResultDto data)
```
please rename to fill marks

BobTheSuperman replied:
Done!

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+            {
+                var dateData = data.StudentsMarks.GroupBy(x => x.LessonId);
+
+                string[] headers = new string[dateData.Count() + 2];
```
please leave a comment

BobTheSuperman replied:
Done!

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+                    rows[i] = new string[headers.Length];
+                    rows[i][0] = students.ElementAt(i);
+
+                    List<int> listOfMarks = new List<int>();
```
please rename

BobTheSuperman replied:
Done

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+
+                        if(rows[i][j + 1] != " "  && rows[i][j + 1] != "-")
+                        {
+                            listOfMarks.Add(Convert.ToInt32(rows[i][j + 1]));
```
please rework according to conversation

BobTheSuperman replied:
Done!

ArsenMaistryshyn commented on CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/ClassbookExportHTML.cs:
```diff
@@ -0,0 +1,167 @@
+                finishDate = data.StudentsMarks.Last().LessonDate?.ToString("dd.MM.yyyy");
+            }
+
+            return $"Group: {group} {startDate}&nbsp;-&nbsp;{finishDate}";
```
please move to constants

BobTheSuperman replied:
Done!

BobTheSuperman committed:
Removed empty spaces and renamed "try to fill" methods to just "fill" methods.
Reworked FillMarks method according to conversation and added some cosmetic changes in HtmlGenarator class.

BobTheSuperman committed:
Created class for response messages and added constant strings for response.
Added some comments for understanding logic in FillMarks method.
Added constants in StudentGroupExportHTML for column numbers.
